lib/log.c

```diff
@@ -85,8 +85,29 @@ static void write_wrapper (int fd, const void *buf, size_t count)
   return;
 }
 
-/* For time string format. */
+/**
+ * Looks up a message in a message list by key.
+ *
+ * If the message is not found, returns the provided error message.
+ *
+ * @param mz the message list
+ * @param kz the message key
+ * @return the message
+ */
+const char *
+lookup_msg(const struct message *mz, int kz, const char *nf)
+{
+  const char *rz = nf ? nf : "(no message found)";
+  const struct message *pnt;
+  for (pnt = mz; pnt->key != 0; pnt++)
+    if (pnt->key == kz) {
+      rz = pnt->str;
```

Comment:
Perhaps adding a test to check that pnt->str is not null will also be safer in case of the message list was not correctly fulfil like that:
rz = pnt->str ? pnt->str : rz;
is too much or not ?